File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{

    //program only runs if there are 2 command line arguments
    if (argc == 2)
    {

        //getting the key (k) from the second command line argument (argv[1]) as an integer
        string s = argv[1];
        int k = atoi (s);

        string plain = get_string("plaintext: ");


        //iterating the cypher over every letter of the string, however long it is (strlen)
        for (int chara = 0; chara < strlen(plain); chara ++)
        {

            //only running the cyper if the character is a letter
            if (isalpha(plain[chara]))
            {
                //formula for upper case letters
                if (isupper(plain[chara]))
                {
                    plain[chara] = ((plain[chara] - 65 + k) % 26) + 65;
                }

                //formula for lower case letters
                else if (islower(plain[chara]))
                {
                    plain[chara] = ((plain[chara] - 97 + k) % 26) + 97;
                }
            }

            //if character isn't a letter, leaves it as is
            else
            {
                plain[chara] = plain[chara];
            }
        }

        printf("ciphertext: %s\n", plain);
        return 0;
    }

    else
    {
        printf("SMH THATS AN ERROR\n");
    }
    return 1;
}

File: caesar.py
from cs50 import get_string
import sys

key = int(sys.argv[1])

# making sure that the correct number of command line arguments exists
if (len(sys.argv) != 2):
    print("error")
    exit(1)

# prompting user for plaintext string
print("plaintext: ", end="")
string = get_string()
string2 = ""

for i in range(len(string)):

    if string[i].isalpha():

        # algoritm for enciphering uppercase letters
        if string[i].isupper():
            string2 += chr(((ord(string[i]) - 65 + key) % 26) + 65)

        # algorithm for enciphering lowercase letters
        if string[i].islower():
            string2 += chr(((ord(string[i]) - 97 + key) % 26) + 97)

    else:
        string2 += string[i]

print("ciphertext:", string2)
